File: app/src/main/java/com/rdd/finy/data/WalletDao.java
package com.rdd.finy.data;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.*;

import java.util.List;

@Dao
public interface WalletDao {

    @Query("SELECT * FROM Wallet WHERE id=:id")
    LiveData<Wallet> findById(int id);

    @Query("SELECT * FROM Wallet")
    LiveData<List<Wallet>> findAll();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(Wallet wallet);

    @Delete
    int delete(Wallet wallet);

}

File: app/src/main/java/com/rdd/finy/data/WalletDao.kt
package com.rdd.finy.data

import androidx.room.*
import io.reactivex.Flowable
import io.reactivex.Single

@Dao
interface WalletDao {

    @Query("SELECT * FROM Wallet WHERE id=:id")
    fun getWalletById(id: Long): Single<Wallet>

    @Query("SELECT * FROM Wallet")
    fun getAllWallets(): Flowable<List<Wallet>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(wallet: Wallet): Long

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun update(wallet: Wallet): Int

    @Delete
    fun delete(wallet: Wallet): Int

}
